packages/nuxt/src/app/components/nuxt-link.ts

```diff
@@ -299,10 +316,26 @@ export function defineNuxtLink (options: NuxtLinkOptions) {
       const el = import.meta.server ? undefined : ref<HTMLElement | null>(null)
       const elRef = import.meta.server ? undefined : (ref: any) => { el!.value = props.custom ? ref?.$el?.nextElementSibling : ref?.$el }
 
+      function checkShouldPrefetch () {
+        const shouldPrefetch = props.prefetch !== false && props.noPrefetch !== true && props.target !== '_blank' && !isSlowConnection() && !prefetched.value
+        return shouldPrefetch
+      }
```

Comment:
```suggestion
function shouldPrefetch (): boolean {
        return props.prefetch && !props.noPrefetch && props.target !== '_blank' && !isSlowConnection() && !prefetched.value
      }
```

Reply:
@quentinmcq the `check` prefix is more explicit, `shouldPrefetch` as a function, does not resonate with me. `checkShouldPrefetch` is clearer and depict the fact that this is a function and calling it I'm checking if prefetch should be done, adding return type is useless, inline return without variable looks like something worth considering

Reply:
> @quentinmcq the check prefix is more explicit, shouldPrefetch as a function, does not resonate with me. checkShouldPrefetch is clearer and depict the fact that this is a function and calling it I'm checking if prefetch should be done, adding return type is useless, inline return without variable looks like something worth considering

I can't find this comment but to answer it, the fact is that the name of your function is supposed to concretely define what it does there. If you add a const behind just for a return line, your function is poorly named and in our case it is redundant. Your function returns a boolean regarding whether it should retrieve data or not, so "should prefetch" matches perfectly, no need for a prefix.

Reply:
@quentinmcq yeah at first glance it was the case but then after thinking more I agree with you, so I removed the comment and implemented suggested change, thx